src/spyglass/utils/mixins/ingestion.py

```diff
@@ -394,22 +374,34 @@ def validate1_duplicate(self, tbl, new_key):
             The table to validate against.
         new_key : dict
             The new key to validate against existing entries in the database.
+
+        Returns
+        -------
+        dict or None
+            The new entry to insert after validation, or None if the entry
+            already exists and is consistent.
         """
         # NOTE: switching from `self` to `tbl` to allow validation from
         # FreeTable instances captured with `parts(as_objects=True)`
+        adjusted_entries = tbl._adjust_keys_for_entry([new_key])
+        if not adjusted_entries:
+            return  # entry filtered out by adjustment
 
-        # If novel entry, nothing to validate
-        if not (query := tbl & new_key):
-            return
+        adj_new_key = adjusted_entries[0]
+        primary_key = {
+            k: v for k, v in adj_new_key.items() if k in tbl.primary_key
+        }
+        if not (query := (tbl & primary_key)):
+            return new_key  # If novel primary key, nothing to validate
```

Comment:
Shouldn't we use the adjusted version for entry, in case it's different?
```suggestion
            return adj_new_key  # If novel primary key, nothing to validate
```